pvlib/tests/test_irradiance.py

```diff
@@ -1095,6 +1095,29 @@ def test_dirindex(times):
         equal_nan=True)
 
 
+@fail_on_pvlib_version("0.13")
+def test_dirindex_ghi_clearsky_deprecation():
+    with pytest.warns(pvlibDeprecationWarning, match='ghi_clear'):
+        times = pd.DatetimeIndex(['2014-06-24T18-1200'])
+        ghi = pd.Series([1038.62], index=times)
+        ghi_clearsky = pd.Series(
+            np.array([1042.48031487]),
+            index=times
+        )
```

Comment:
Nitpick: it's best practice to include only the warning-emitting code inside the `with pytest.warns(...):` block.  Reasons: it makes the intent of the test clearer (future readers don't have to figure out which line of code is supposed to be emitting the warning), and it theoretically makes the test more robust (technically, it is possible that `pandas` could start emitting `pvlibDeprecationWarnings` talking about `ghi_clear`...).

So let's move all this variable initialization code outside the `pytest.warns` block.

Reply:
Also I'm guessing much of this was just copy/paste from other tests, but no need for the `np.array`s here; `pd.Series([1042.4...], index=times)` should work fine.